cli/compose/loader/loader.go

```diff
@@ -41,11 +52,23 @@ func ParseYAML(source []byte) (map[string]interface{}, error) {
 }
 
 // Load reads a ConfigDetails and returns a fully loaded configuration
-func Load(configDetails types.ConfigDetails) (*types.Config, error) {
+func Load(configDetails types.ConfigDetails, ops ...func(*Options)) (*types.Config, error) {
 	if len(configDetails.ConfigFiles) < 1 {
 		return nil, errors.Errorf("No files specified")
 	}
 
+	opts := &Options{
+		Interpolate: &interp.Options{
+			Substitute:      template.Substitute,
+			LookupValue:     configDetails.LookupEnv,
+			TypeCastMapping: interpolateTypeCastMapping,
+		},
+	}
+
+	for _, op := range ops {
+		op(opts)
```

Comment:
This seems to be the standard pattern, but was just thinking why functional options cannot produce an error 🤔

Reply:
@thaJeztah they could :wink: I just though they would not here :stuck_out_tongue_closed_eyes: